utils/src/main/java/se/fortnox/reactivewizard/util/FieldSetter.java

```diff
@@ -2,29 +2,30 @@
 
 import java.lang.reflect.Field;
 import java.lang.reflect.Type;
+import java.util.function.BiConsumer;
 
 /**
  * Represents a setter field.
  */
-public class FieldSetter implements Setter {
-    public static Setter create(Class<?> cls, Field field) {
+public class FieldSetter<I,T> implements Setter<I,T> {
+    public static <I,T> Setter<I,T> create(Class<I> cls, Field field) {
         AccessorUtil.MemberTypeInfo memberTypeInfo = AccessorUtil.fieldTypeInfo(cls, field);
-        return new FieldSetter(field, memberTypeInfo.getReturnType(), memberTypeInfo.getGenericReturnType());
+        return new FieldSetter<>(field, memberTypeInfo.getReturnType(), memberTypeInfo.getGenericReturnType());
     }
 
     private final Field    field;
     private final Class<?> parameterType;
     private final Type     genericParameterType;
 
-    private FieldSetter(Field field, Class<?> parameterType, Type genericParameterType) {
+    private FieldSetter(Field field, Class<T> parameterType, Type genericParameterType) {
         this.field = field;
         this.parameterType = parameterType;
         this.genericParameterType = genericParameterType;
         field.setAccessible(true);
```

Comment:
Should we for performance and to be consistent create a lambda function in the constructor here as well and use that lambda in the methods invoke and setterFunction. Otherwise we will return a new function each time setterFunction is used right?

Reply:
Good catch! Fixed.